packages/accounts/src/light-account/account.ts

```diff
@@ -32,24 +32,24 @@ import {
 } from "./utils.js";
 
 export type LightAccount<
-  TOwner extends SmartAccountSigner = SmartAccountSigner
-> = OwnedSmartContractAccount<"LightAccount", TOwner> & {
+  TSigner extends SmartAccountSigner = SmartAccountSigner
+> = SmartContractAccountWithSigner<"LightAccount", TSigner> & {
   getLightAccountVersion: () => Promise<LightAccountVersion>;
   encodeTransferOwnership: (newOwner: Address) => Hex;
   getOwnerAddress: () => Promise<Address>;
-  setOwner: <TOwner extends SmartAccountSigner = SmartAccountSigner>(
-    newOwner: TOwner
+  setSigner: <TSigner extends SmartAccountSigner = SmartAccountSigner>(
+    signer: TSigner
   ) => void;
```

Comment:
While we're doing this refactor, can we remove the `setOwner` method? I kept it as mutable to match old behaviour, but mutability in this new context can result in some unexpected behaviour. it also makes building new features more difficult especially if you're using the signer of the account at all

Reply:
this method is actually needed for typescript. yeah I know, was very weird, but get "[Type SmartAccountSigner  is assignable to the constraint of type 'TSigner', but 'TSigner' could be instantiated with a different subtype of constraint 'SmartAccountSigner']"
 
 Bascially, LightAccount needs to use the generic type TSigner in its interface in order for ts to not error.

Reply:
ah misunderstood the comment. okay will remove setSigner but due to above, might need to override getSigner instead.

Reply:
@moldy530 actually we use setSigner method here https://github.com/alchemyplatform/aa-sdk/blob/a08ff18059cd504ae5fff2645b09d64acd7ea39d/packages/accounts/src/light-account/actions/transferOwnership.ts#L58

Reply:
chatted offline, but I think we should remove the `setSigner` from `transferOwnership` altogether and either:
1) update the docs to reflect that on successful UO, devs need to then call `createLightAccount` to get a new instance connected to the new client
OR
2) update the return type for `transferOwnership` to be something like `{hash: Hash; account: LightAccount}` to make it easier for devs